test/Conversion/CalyxToFSM/lower-invoke.mlir

```diff
@@ -0,0 +1,127 @@
+// CHECK-NEXT:   fsm.state @seq_1_while_entry output {
+// CHECK-NEXT:     fsm.output
+// CHECK-NEXT:   } transitions {
+// CHECK-NEXT:     fsm.transition @seq_1_while_seq_0_invoke_id_0
+// CHECK-NEXT:   }
+// CHECK-NEXT:   fsm.state @seq_1_while_seq_2_incr output {
+// CHECK-NEXT:     calyx.enable @incr
+// CHECK-NEXT:     fsm.output
+// CHECK-NEXT:   } transitions {
+// CHECK-NEXT:     fsm.transition @seq_1_while_header
+// CHECK-NEXT:   }
+// CHECK-NEXT:   fsm.state @seq_1_while_seq_1_invoke_r_0 output {
+// CHECK-NEXT:     calyx.enable @invoke_r_0
+// CHECK-NEXT:     fsm.output
+// CHECK-NEXT:   } transitions {
+// CHECK-NEXT:     fsm.transition @seq_1_while_seq_2_incr
+// CHECK-NEXT:   }
+// CHECK-NEXT:   fsm.state @seq_1_while_seq_0_invoke_id_0 output {
+// CHECK-NEXT:     calyx.enable @invoke_id_0
+// CHECK-NEXT:     fsm.output
+// CHECK-NEXT:   } transitions {
+// CHECK-NEXT:     fsm.transition @seq_1_while_seq_1_invoke_r_0
+// CHECK-NEXT:   }
+// CHECK-NEXT:   fsm.state @seq_0_init output {
+// CHECK-NEXT:     calyx.enable @init
+// CHECK-NEXT:     fsm.output
+// CHECK-NEXT:   } transitions {
+// CHECK-NEXT:     fsm.transition @seq_1_while_header
+// CHECK-NEXT:   }
+// CHECK-NEXT: }
+
+module attributes {calyx.entrypoint = "main"} { 
+calyx.component @identity(%in: i32, %go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%out: i32, %done: i1 {done}) {
+  %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1
+  calyx.wires {
+    calyx.assign %out = %r.out :i32
+  }
+  calyx.control {
+    calyx.seq {
+      calyx.invoke @r(%r.in = %in) -> (i32)
+    }
+  }
+}
+
+calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%done: i1 {done}, %out : i32) {
+  %id.in, %id.go, %id.clk, %id.reset, %id.out, %id.done = calyx.instance @id of @identity : i32, i1, i1, i1, i32, i1
+  %counter.in, %counter.write_en, %counter.clk, %counter.reset, %counter.out, %counter.done = calyx.register @counter : i32, i1, i1, i1, i32, i1
+  %add.left, %add.right, %add.out = calyx.std_add @add : i32, i32, i32
+  %lt.left, %lt.right, %lt.out = calyx.std_lt @lt : i32, i32, i1
+  %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1 
+  %c0 = hw.constant 0 : i32
+  %c1 = hw.constant 1 : i32
+  %c8 = hw.constant 8 : i32
+  %c10 = hw.constant 10 : i32
+  %true = hw.constant 1 : i1
+  calyx.wires {
+    calyx.group @init {
+       calyx.assign %counter.in = %c0 : i32
+       calyx.assign %counter.write_en = %true : i1
+       calyx.group_done %counter.done : i1
+      }
+      calyx.group @incr {
+        calyx.assign %add.left = %counter.out : i32
+        calyx.assign %add.right = %c1 : i32
+        calyx.assign %counter.in = %add.out : i32
+        calyx.assign %counter.write_en = %true : i1
+        calyx.group_done %counter.done : i1
+      }
+      calyx.assign %lt.left = %counter.out : i32
+      calyx.assign %lt.right = %c8 : i32
+    }
```

Comment:
@rachitnigam  I'd like to ask you a question,at first I wrote it in the following form.
```
      calyx.comb_group @cond {
        calyx.assign %lt.left = %counter.out : i32
        calyx.assign %lt.right = %c8 : i32
      }
    }
    calyx.control {
      calyx.seq {
        calyx.enable @init
        calyx.while %lt.out with @cond {
          calyx.seq {
            calyx.invoke @id(%id.in = %c10) -> (i32)
            calyx.invoke @r(%r.in = %id.out) -> (i32)
            calyx.enable @incr
          }
        }
      }
    }
  }
```
When I run it it reports an error.
```
38:7: error: 'calyx.comb_group' op with name: "cond" is unused in the control execution schedule
      calyx.comb_group @cond {
      ^
38:7: note: see current operation: 
"calyx.comb_group"() ({
  "calyx.assign"(%3#0, %1#4) : (i32, i32) -> ()
  "calyx.assign"(%3#1, %7) : (i32, i32) -> ()
}) {sym_name = "cond"} : () -> ()
```
But it does work on calyx's native compiler, I'm a bit confused why it doesn't work here?

Reply:
This is probably a compiler bug in CIRCT.

https://github.com/llvm/circt/blob/73593b923ffe3a47ab75e3e60200244811ee31b8/lib/Dialect/Calyx/CalyxOps.cpp#L1004-L1015

I think https://github.com/llvm/circt/pull/5645 will fix this.

Reply:
Ah nevermind, I was mistaken. From your code sample, it seems the symbol for the `while` condition isn't being added. I would open an issue with a minimal repro test case and command of this.